daprdocs/content/en/developing-applications/building-blocks/state-management/howto-encrypt-state.md

```diff
@@ -89,7 +97,7 @@ metadata:
 
 When Dapr starts, it will fetch the secrets containing the encryption keys listed in the `metadata` section. Dapr knows which state item has been encrypted with which key automatically, as it appends the `secretKeyRef.name` field to the end of the actual state key.
 
-To rotate a key, simply change the `primaryEncryptionKey` to point to a secret containing your new key, and move the old primary encryption key to the `secondaryEncryptionKey`. New data will be encrypted using the new key, and old data that's retrieved will be decrypted using the secondary key. Any updates to data items encrypted using the old key will be re-encrypted using the new key.
+To rotate a key, change the `primaryEncryptionKey` to point to a secret containing your new key, and move the old primary encryption key to the `secondaryEncryptionKey`. New data will be encrypted using the new key, and old data that's retrieved will be decrypted using the secondary key. Any updates to data items encrypted using the old key will be re-encrypted using the new key.
```

Comment:
I do feel that we should provide an appropriate warning here that you need update all the encrpyted state yourself otherwise you may loose the key if you then did another key rotation (i.e. created a third key to rotate to). @ItalyPaleAle - What do you think?

Reply:
There's an open discussion on that. @artursouza  is proposing adding support for multiple keys. See: https://github.com/dapr/dapr/issues/4341

But yes, you are correct that today it's not really a "key rotation"  but rather switching keys. Users are on their own to re-encrypt all existing data. I think a warning would be helpful to add, but then we should also add sample code for doing that? (Which could lead to people deciding to re-encrypt large number of records...)

Reply:
At least make people aware of this since once they go down the path of using the encryption then they cannot easily go back. So let's add a paragraph on this so that this is explicit.

Reply:
Updated